libs/openFrameworks/gl/ofShader.cpp

```diff
@@ -381,29 +427,13 @@ void ofShader::checkProgramInfoLog(GLuint program) {
 	if (infoLength > 1) {
 		GLchar* infoBuffer = new GLchar[infoLength];
 		glGetProgramInfoLog(program, infoLength, &infoLength, infoBuffer);
+		// TODO: it appears that Raspberry Pi only reports shader errors whilst linking,
+		// but then it becomes hard to figure out whether the fragment or the
+		// vertex shader caused the error.
+		// We need to find a robust way of extracing this information from
+		// the log, and unfortunately can't use regex whilst gcc on RPi is assumed to
+		// be < 4.9, which is the first version fully supporting this c++11 feature.
 		string msg = "ofShader: program reports:\n";
-#ifdef TARGET_RASPBERRYPI
```

Comment:
let's leave this in but ifdef'd for HAS_REGEX. you can run arch linux in the pi which already is on gcc 4.9 or 5